singularity-opac/neutrinos/gray_s_opacity_neutrinos.hpp

```diff
@@ -0,0 +1,71 @@
+// ======================================================================
+// © 2022. Triad National Security, LLC. All rights reserved.  This
+// program was produced under U.S. Government contract
+// 89233218CNA000001 for Los Alamos National Laboratory (LANL), which
+// is operated by Triad National Security, LLC for the U.S.
+// Department of Energy/National Nuclear Security Administration. All
+// rights in the program are reserved by Triad National Security, LLC,
+// and the U.S. Department of Energy/National Nuclear Security
+// Administration. The Government is granted for itself and others
+// acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works,
+// distribute copies to the public, perform publicly and display
+// publicly, and to permit others to do so.
+// ======================================================================
+
+#ifndef SINGULARITY_OPAC_NEUTRINOS_GRAY_S_OPACITY_NEUTRINOS_
+#define SINGULARITY_OPAC_NEUTRINOS_GRAY_S_OPACITY_NEUTRINOS_
+
+#include <cassert>
+#include <cmath>
+#include <cstdio>
+
+#include <ports-of-call/portability.hpp>
+#include <singularity-opac/base/opac_error.hpp>
+
+namespace singularity {
+namespace neutrinos {
+
+template <typename pc = PhysicalConstantsCGS>
+class GraySOpacity {
+ public:
+  GraySOpacity() = default;
+  GraySOpacity(const Real sigma, const Real mmw) : sigma_(sigma), mmw_(mmw) {}
```

Comment:
Unlike on the absorption opacity side now we're dealing with both scattering coefficients and cross sections, so I think a reasonable solution right now is to pass mean molecular weight `mmw` to constructors generally to convert between mass density and number density. If MMW varies inside a calculation (due to Ye in NS mergers?) we may need to revisit this model.

Reply:
is "molecular" weight the right concept here? More like molar weight or just average particle mass? Also it does for sure change for NS mergers, but it's incorporated already in opacity tables for the most part, which assume an EOS

Reply:
"mean molecular weight" is typical in astrophysics even for this sort of thing I think, but since these indeed aren't typically molecules I switched to your `avg_particle_mass`.

OK when we add tabulated NS scattering opacities we can just leverage the data appropriately to return both `sigma` and `kappa` rather than providing an `avg_particle_mass` value to the constructor.